### PR TITLE
backend-storage: support (and prefer) the default storage class for virtualization

### PR DESCRIPTION
**Pull request by jean-edouard**
### What this PR does
Before this PR:
backend-storage defaults to using the default storage class for the cluster

After this PR:
backend-storage defaults to using the default storage class for virtualization workloads, then falls back to the cluster default

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve

File: pkg/storage/backend-storage/backend-storage.go [hunk 1]
```diff
@@ -98,14 +98,25 @@ func (bs *BackendStorage) getStorageClass() (string, error) {
...
 	}
 
-	return "", fmt.Errorf("no default storage class found")
+	if kvDefault != "" {
+		return kvDefault, nil
+	} else if k8sDefault != "" {
+		return k8sDefault, nil
+	} else {
+		return "", fmt.Errorf("no default storage class found")
+	}
```
Comment (fossedihelm) [inline pkg/storage/backend-storage/backend-storage.go:119]: Not a fan of if / else if / else if :) 

IDK just an alternative as proposal, don't really know if this is better:
```
	k8sDefault := ""
	for _, obj := range bs.scStore.List() {
		sc := obj.(*storagev1.StorageClass)
		if sc.Annotations["storageclass.cdi.kubevirt.io/is-default-class"] == "true" {
			return sc.Name, nil
		}
		if sc.Annotations["storageclass.kubernetes.io/is-default-class"] == "true" {
			k8sDefault = sc.Name
		}
	}

	if k8sDefault != "" {
		return k8sDefault, nil
	}
	return "", fmt.Errorf("no default storage class found")
```

Comment (fossedihelm) [reply pkg/storage/backend-storage/backend-storage.go:119]: Feel free to choose the one you prefer

**Review by fossedihelm — APPROVED** [review]
A unit test would be great!

File: pkg/storage/backend-storage/backend-storage.go (hunk 1 above)
Comment (jean-edouard) [reply pkg/storage/backend-storage/backend-storage.go:119]: My first version actually looked exactly like the above :)
I went with the if / else if / else approach instead to highlight what the algorithm does: "pick the virt class if it exists, else pick the k8s one if it exists, else fail".
It also feels too trivial for a unit test...
Please hold/unlgtm if you feel strongly about either.
Thanks for the review!

Comment (fossedihelm) [reply pkg/storage/backend-storage/backend-storage.go:119]: That's fine as is :)

**Pull request merged**